EventProcessor/EventProcessor.WebJob/Processors/MessageFeedbackProcessor.cs

```diff
@@ -80,116 +73,88 @@ public void Stop()
-                        device = 
-                            await _deviceLogic.GetDeviceAsync(record.DeviceId);
-                        if (device == null)
-                        {
-                            continue;
-                        }
-
-                        Trace.TraceInformation(
+                        UpdateDeviceRecord(record, batch.EnqueuedTime);
+                    }
+
+                    await batchReceiver.CompleteAsync(batch);
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.TraceError("Error in MessageFeedbackProcessor.RunProcess, Exception: {0}", ex.Message);
+            }
+            finally
+            {
+                if (serviceClient != null)
+                {
+                    await serviceClient.CloseAsync();
+                }
+
+                _isRunning = false;
+            }
+            
+        }
+
+        private async void UpdateDeviceRecord(FeedbackRecord record, DateTime enqueuDateTime)
+        {
+            Trace.TraceInformation(
                             "{0}{0}*** Processing Feedback Record ***{0}{0}DeviceId: {1}{0}OriginalMessageId: {2}{0}Result: {3}{0}{0}",
                             Console.Out.NewLine,
                             record.DeviceId,
                             record.OriginalMessageId,
                             record.StatusCode);
 
-                        existingCommand =
-                            CommandHistorySchemaHelper.GetCommandHistoryItemOrDefault(
-                                device,
-                                record.OriginalMessageId);
-
-                        if (existingCommand == null)
-                        {
-                            continue;
-                        }
-
-                        updatedTime = record.EnqueuedTimeUtc;
-                        if (updatedTime == default(DateTime))
-                        {
-                            updatedTime = batch.EnqueuedTime;
-                        }
-
-                        if (updatedTime == default(DateTime))
-                        {
-                            updatedTime = DateTime.UtcNow;
-                        }
-
-                        existingCommand.UpdatedTime = updatedTime;
-
-                        existingCommand.Result = record.StatusCode.ToString();
-
-                        if (record.StatusCode == FeedbackStatusCode.Success)
-                        {
-                            existingCommand.ErrorMessage = string.Empty;
-                        }
-
-                        CommandHistorySchemaHelper.UpdateCommandHistoryItem(
-                            device, 
-                            existingCommand);
-
-                        await _deviceLogic.UpdateDeviceAsync(device);
-                    }
-
-                    await batchReceiver.CompleteAsync(batch);
-                }
+            var device = await _deviceLogic.GetDeviceAsync(record.DeviceId);
```

Comment:
See the line below that uses a null terminator. So if device is null or existing command is null it will return